pandas/core/series.py

```diff
@@ -2132,10 +2133,15 @@ def map_f(values, f):
             else:
                 map_f = lib.map_infer
 
-        if isinstance(arg, (dict, Series)):
-            if isinstance(arg, dict):
-                arg = self._constructor(arg, index=arg.keys())
+        default_dict_types = collections.Counter, collections.defaultdict
```

Comment:
instead of actually naming these, use ``issubclass(type(arg), dict) and not type(arg) is dict``

```
In [17]: f = lambda arg: issubclass(type(arg), dict) and not type(arg) is dict

In [18]: f({})
Out[18]: False

In [19]: f(collections.defaultdict())
Out[19]: True

In [21]: f(collections.Counter())
Out[21]: True

```